src/libaktualizr/primary/aktualizr.cc

```diff
@@ -43,33 +43,54 @@ void Aktualizr::systemSetup() {
   LOG_TRACE << "... seeding complete in " << timer;
 }
 
-void Aktualizr::Initialize() { uptane_client_->initialize(); }
-
-void Aktualizr::UptaneCycle() {
-  RunningMode running_mode = config_.uptane.running_mode;
-  UpdateCheckResult update_result = CheckUpdates();
-  if (running_mode == RunningMode::kCheck || update_result.updates.size() == 0) {
-    return;
-  } else if (running_mode == RunningMode::kInstall) {
-    Install(update_result.updates);
-    uptane_client_->putManifest();
-    return;
-  }
+void Aktualizr::Initialize() {
+  std::thread([&]() {
+    while (!shutdown_) {
+      auto task = api_queue_.dequeue();
+      if (shutdown_) {
+        return;
+      }
+      task();
+    }
+  })
+      .detach();
+  uptane_client_->initialize();
+}
```

Comment:
We should probably initialize SotaUptaneClient before launching the thread just to be safe.